metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -155,9 +176,35 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         # can't make this method abstract.
         raise NotImplementedError('sources must implement "create"')
 
-    @abstractmethod
+    def get_workunit_processors(self) -> Sequence[Optional[MetadataWorkUnitProcessor]]:
```

Comment:
I think we should change this return type to `List[Optional[MetadataWorkUnitProcessor]]`

Reply:
So we can `append` to it? I guess it's more specific.

Reply:
not for append - we shouldn't be using append here

Sequence is a bit weird to work with in some places with mypy. We can always make it more general in the future, but let's not restrain ourselves unnecessarily

Reply:
Oh really, what issues have you seen with Sequence? In general, I think we want to use `Sequence` (if possible) for parameters but `List` / other specific types for return types

Reply:
Actually i think sequences are fine. iirc the issue i've seen is that Iterable is not covariant (so passing Iterable[Square] to a function that takes Iterable[Shape] would throw an error), but sequence works fine there

> In general, I think we want to use Sequence (if possible) for parameters but List / other specific types for return types

yup this seems like a good rule of thumb - might be worth adding to the list here https://datahubproject.io/docs/metadata-ingestion/developing/#code-style